tracer/test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -822,21 +855,30 @@ private void HandleHttpRequests()
                                 ctx.Response.AddHeader("Datadog-Agent-Version", Version);
                             }
 
-                            var response = HandleHttpRequest(MockHttpParser.MockHttpRequest.Create(ctx.Request));
-                            var buffer = Encoding.UTF8.GetBytes(response);
-                            ctx.Response.ContentType = "application/json";
-                            ctx.Response.ContentLength64 = buffer.LongLength;
-                            ctx.Response.OutputStream.Write(buffer, 0, buffer.Length);
+                            var mockTracerResponse = HandleHttpRequest(MockHttpParser.MockHttpRequest.Create(ctx.Request));
+
+                            if (mockTracerResponse.SendResponse)
+                            {
+                                var buffer = Encoding.UTF8.GetBytes(mockTracerResponse.Response);
+
+                                if (ctx.Response.StatusCode == 200)
+                                {
+                                    ctx.Response.StatusCode = mockTracerResponse.StatusCode;
+                                }
+
```

Comment:
The problem is that ctx.Response could come with an previous status code different than 200 and we would want to keep that code, otherwise other integration tests would fail.

Reply:
Oh, interesting🤔 I don't see anywhere in the Mock agent that we set it? 🤔 Oh wait, we have callbacks, I remember 👍